src/Agent/NewRelic/Agent/Core/Segments/Segment.cs

```diff
@@ -219,7 +220,33 @@ private void Finish()
         {
             var endTime = _transactionSegmentState.GetRelativeTime();
             RelativeEndTime = endTime;
-            _parameters = Data.Finish() ?? EmptyImmutableParameters;
+            _parameters = Data.Finish();
+            using (var configurationSubscription = new ConfigurationSubscriber())
```

Comment:
I'm thinking each class should just instantiate one static instance, if config access is needed